### PR TITLE
Change source of TLS certs + Discover OpenShift routes external addresses

### PR DESCRIPTION
**Pull request by nb-ohad**
### Explain the changes
1. Replace NooBaa's cert mechanism to use a mounted k8s secrets
2. Discover external addresses service by OpenShift routes

### Issues: Fixed #xxx / Gap #xxx
1. 

### Testing Instructions:
1.

### CODE REVIEW

File: src/server/system_services/system_server.js [hunk 1]
```diff
@@ -1020,6 +959,22 @@ function _list_s3_endpoints(system) {
         });
 }
 
+async function using_local_certs() {
```
Comment (nimrod-becker) [inline src/server/system_services/system_server.js]: why not in ssl_utils?

**Review by nimrod-becker — APPROVED** [review]

File: src/server/system_services/system_server.js (hunk 1 above)
Comment (nb-ohad) [reply src/server/system_services/system_server.js]: will do

File: .travis/travis_test_job.yaml [hunk 2]
```diff
@@ -15,13 +15,13 @@ spec:
...
+
           - node
           - ./src/test/framework/test_env_builder_kubernetes.js
           - --image
-          - NOOBAA_IMAGE_PLACEHOLDER
-          - --namespace_prefix 
-          - NAMESPACE_PREFIX_PLACEHOLDER
+          - "NOOBAA_IMAGE_PLACEHOLDER"
+          - --namespace_prefix
+          - "NAMESPACE_PREFIX_PLACEHOLDER"
```
[the remark this answers is not in the document]
Comment (nimrod-becker) [reply .travis/travis_test_job.yaml]: on purpose since it's the hash and it might start with a number

[the remark this answers is not in the document]
Comment (nb-ohad) [reply .travis/travis_test_job.yaml]: This value is a placeholder that is replaced with the 7 first chars of a commit hash which can be only digits. in that case, we want to guarantee that the value will be interrupted as a string.

File: src/util/os_utils.js [hunk 3]
```diff
@@ -797,6 +825,49 @@ function sort_address_list(address_list) {
     });
 }
 
+async function restart_services(services) {
```
Comment (nimrod-becker) [inline src/util/os_utils.js]: please take a look at supervisor_ctrl.js

**Review by nimrod-becker — APPROVED** [review]

File: src/util/ssl_utils.js [hunk 4]
```diff
@@ -27,26 +32,72 @@ function verify_ssl_certificate(certificate) {
...
-        });
+    );
+}
+
+// Overide the current loaded certs with new certs.
+async function update_ssl_certificates(new_certs) {
+    dbg.log0(`update_ssl_certificates: Updating loaded certificate for ${Object.keys(new_certs)}`);
+    using_generated_certs = false;
+    certs = {
+        ...(certs || {}),
```
Comment (guymguym) [inline src/util/ssl_utils.js]: Why do you merge the objects? Why not replace?
Even if there is a reason, you have to write a comment to explain it as it is not trivial to see it when reading.

File: src/util/ssl_utils.js [hunk 5]
```diff
@@ -27,26 +32,72 @@ function verify_ssl_certificate(certificate) {
...
+        ...new_certs
+    };
+}
+
+async function is_using_local_certs() {
```
Comment (guymguym) [inline src/util/ssl_utils.js]: I don't understand the terminology of "local" certificates.
Why not `function is_using_generated_certs() { return using_generated_certs; }`?

File: src/util/ssl_utils.js [hunk 6]
```diff
@@ -27,26 +32,72 @@ function verify_ssl_certificate(certificate) {
...
+    return P.props(
+        _.mapValues(SSL_CERTS_DIR_PATHS, async dir => {
+            const key_path = path.join(dir, 'tls.key');
+            const cert_path = path.join(dir, 'tls.crt');
+            const certificate = {
+                key: await fs.promises.readFile(key_path, 'utf8'),
+                cert: await fs.promises.readFile(cert_path, 'utf8')
+            };
+
             verify_ssl_certificate(certificate);
```
Comment (guymguym) [inline src/util/ssl_utils.js]: Not sure it's clear what is the behavior when the provided certificate file is invalid - do we fallback to generated cert? or do we crash the server?

File: src/util/ssl_utils.js (hunk 5 above)
Comment (nb-ohad) [reply src/util/ssl_utils.js]: The old code used the term local certs, I started from that code so the name matches the old naming convention. 
I will rename in a future commit

**Pull request merged**